src/test/java/spoon/test/eval/EvalTest.java

```diff
@@ -465,13 +465,12 @@ void testVisitCtBinaryOperatorLiteralType(
 			.getElements(new TypeFilter<>(CtBinaryOperator.class))
 			.get(0);
 
-		CtType<?> currentType = ctBinaryOperator.getType().getTypeDeclaration().clone();
 		CtExpression<?> evaluated = ctBinaryOperator.partiallyEvaluate();
 		assertNotNull(
 			evaluated.getType(),
 			String.format("type of '%s' is null after evaluation", ctBinaryOperator)
 		);
-		assertEquals(currentType, evaluated.getType().getTypeDeclaration());
+		assertEquals(ctBinaryOperator.getType(), evaluated.getType());
```

Comment:
The test should only check for the type of the binary operator and the evaluated result. There is no need to check the equality between type declarations.